src/Sarif.UnitTests/SarifCurrentToVersionOneTests.cs

```diff
@@ -0,0 +1,204 @@
+﻿// Copyright (c) Microsoft. All rights reserved. Licensed under the MIT        
+// license. See LICENSE file in the project root for full license information.
+
+using FluentAssertions;
+using Microsoft.CodeAnalysis.Sarif.Readers;
+using Microsoft.CodeAnalysis.Sarif.VersionOne;
+using Microsoft.CodeAnalysis.Sarif.Visitors;
+using Newtonsoft.Json;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Sarif
+{
+    public class SarifCurrentToVersionOneTests
+    {
+        private static readonly JsonSerializerSettings s_v1JsonSettings = new JsonSerializerSettings
+        {
+            ContractResolver = SarifContractResolverVersionOne.Instance,
+            Formatting = Formatting.Indented
+        };
+
+        private static readonly JsonSerializerSettings s_v2JsonSettings = new JsonSerializerSettings
+        {
+            ContractResolver = SarifContractResolver.Instance,
+            Formatting = Formatting.Indented
+        };
+
+        private static SarifLog GetSarifLog(string logText)
+        {
+            return JsonConvert.DeserializeObject<SarifLog>(logText, s_v2JsonSettings);
+        }
+
+        private static SarifLogVersionOne TransformCurrentToVersionOne(string v2LogText)
+        {
+            SarifLog v2Log = GetSarifLog(v2LogText);
+            var transformer = new SarifCurrentToVersionOneVisitor();
+            transformer.VisitSarifLog(v2Log);
+
+            return SarifCurrentToVersionOneVisitor.SarifLogVersionOne;
```

Comment:
I should have mentioned this in the last PR. The `SarifLogVersionOne` property is the one thing you _don't_ want to be static. Consider a multithreaded application that creates two transformers on two threads. The application tells one of the transformers to transform v2LogOne, and the other to transform v2LogTwo. Both transformer objects use the same set of static methods to build up two _different_ output objects.

Reply:
Yeah understood. The trouble there is that I have two spots that refer to it directly, so I'll have to refactor.

Reply:
So I'm not sure how to make this work: in CreateInvocation, I'm passing the last run we created to CreateResponseFilesList in case there are responseFiles that need to be added to the run's files collection. I'm not coming up with any slick ideas on how to get at that run instance, short of passing it all the way through as a parameter. Thoughts?